### PR TITLE
chore(broker-core): start event uses event scopes

### LINKED ISSUE #2128: Message start event subscription uses event scope instance state
* Create an EventScopeInstance for workflows that have message start events (scopeKey is the workflowKey, accepting=true, no interrupting ids)
* Put payload in EventScopeInstanceState and remove it from record
* When handling the event occurred set the payload in the variable state payload column (with the scope key of the start event)

### PR DESCRIPTION
**Pull request by miguelpires**
closes #2128

### CODE REVIEW

File: broker-core/src/main/java/io/zeebe/broker/subscription/message/processor/MessageEventProcessors.java [hunk 1]
```diff
@@ -46,11 +46,7 @@ public static void addMessageProcessors(
         .onCommand(
             ValueType.MESSAGE,
             MessageIntent.PUBLISH,
-            new PublishMessageProcessor(
-                messageState,
-                subscriptionState,
-                startEventSubscriptionState,
-                subscriptionCommandSender))
+            new PublishMessageProcessor(zeebeState, subscriptionCommandSender))
```
Comment (npepinpe) [inline broker-core/src/main/java/io/zeebe/broker/subscription/message/processor/MessageEventProcessors.java]: I'd rather we pass in only the direct dependencies as opposed to passing the whole parent object; it makes it clearer what the dependencies of the object are, and also makes it a whole lot easier to test/reuse in isolation.

I can do this change myself though, so no worries.

File: broker-core/src/main/java/io/zeebe/broker/subscription/message/processor/MessageEventProcessors.java [hunk 2]
```diff
@@ -70,7 +66,8 @@ public static void addMessageProcessors(
         .onCommand(
             ValueType.MESSAGE_START_EVENT_SUBSCRIPTION,
             MessageStartEventSubscriptionIntent.OPEN,
-            new OpenMessageStartEventSubscriptionProcessor(startEventSubscriptionState))
+            new OpenMessageStartEventSubscriptionProcessor(
+                startEventSubscriptionState, zeebeState.getWorkflowState()))
```
Comment (npepinpe) [inline broker-core/src/main/java/io/zeebe/broker/subscription/message/processor/MessageEventProcessors.java]: Same as above, it's clearer to pass the direct dependency in general.

File: broker-core/src/main/java/io/zeebe/broker/subscription/message/processor/OpenMessageStartEventSubscriptionProcessor.java [hunk 3]
```diff
@@ -44,6 +48,10 @@ public void processRecord(
     final MessageStartEventSubscriptionRecord subscription = record.getValue();
     subscriptionState.put(subscription);
 
+    workflowState
+        .getEventScopeInstanceState()
+        .createIfNotExists(subscription.getWorkflowKey(), Collections.EMPTY_LIST);
```
Comment (npepinpe) [inline broker-core/src/main/java/io/zeebe/broker/subscription/message/processor/OpenMessageStartEventSubscriptionProcessor.java]: `Collections.emptyList()` is the type safe way of using `Collections.EMPTY_LIST`.

File: broker-core/src/main/java/io/zeebe/broker/workflow/processor/handlers/container/ContainerElementActivatedHandler.java [hunk 4]
```diff
@@ -76,8 +77,15 @@ protected boolean handleState(BpmnStepContext<T> context) {
   private void populateRecordFromDeferredRecord(
       BpmnStepContext<T> context, WorkflowInstanceRecord value) {
     final long wfInstanceKey = context.getRecord().getValue().getWorkflowInstanceKey();
+    final EventTrigger eventTrigger =
```
Comment (npepinpe) [inline broker-core/src/main/java/io/zeebe/broker/workflow/processor/handlers/container/ContainerElementActivatedHandler.java]: Not needed

File: broker-core/src/main/java/io/zeebe/broker/workflow/processor/handlers/container/ContainerElementActivatedHandler.java [hunk 5]
```diff
@@ -89,9 +97,9 @@ private void populateRecordFromDeferredRecord(
 
     final IndexedRecord deferredRecord = deferredRecords.get(0);
     final WorkflowInstanceRecord workflowInstanceRecord = deferredRecord.getValue();
-    value.setElementId(workflowInstanceRecord.getElementId());
-    value.setBpmnElementType(workflowInstanceRecord.getBpmnElementType());
-    value.setPayload(workflowInstanceRecord.getPayload());
+    value.setElementId(eventTrigger.getElementId());
+    value.setBpmnElementType(deferredRecord.getValue().getBpmnElementType());
+    value.setPayload(eventTrigger.getPayload());
```
Comment (npepinpe) [inline broker-core/src/main/java/io/zeebe/broker/workflow/processor/handlers/container/ContainerElementActivatedHandler.java]: Can be removed

**Review by npepinpe — CHANGES_REQUESTED** [review]

**Comment by MiguelPires** [comment]
Hey @npepinpe, I addressed your comments. Please have a look when you have time

**Review by npepinpe — APPROVED** [review]

**Pull request merged**